Code/Framework/AzCore/AzCore/Slice/SliceComponent.cpp

```diff
@@ -1061,41 +1064,64 @@ namespace AZ
             return true;
         }
 
-        AZ::Data::AssetInfo assetInfo;
-        AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, m_asset.GetId());
-        bool isAssetStillOnDisk = assetInfo.m_assetId.IsValid();
-
-        bool isCachedAssetReady = m_asset.IsReady();
-
-        if (!(isCachedAssetReady && isAssetStillOnDisk))
+        SliceComponent* dependentSlice = nullptr;
+        const bool useAssetProcessor = !serializeContext || !relativeToAbsoluteSlicePaths;
+        if (useAssetProcessor)
```

Comment:
Well this boolean is really to use the AssetCatalog, not the Asset Processor. I suggest to change the name to useAssetCatalog.

The Asset Processor processes Assets and adds to assetcatlog.xml file in the Cache.
However when the SerializeContextTools application is running, it uses the static Asset Catalog file on disk and doesn't interact with any running Asset Processors in real time.